flutter/lib/src/screenshot/sentry_screenshot.dart

```diff
@@ -22,32 +39,49 @@ class SentryScreenshot extends StatelessWidget {
   }
 }
 
-class ScreenshotAttachment extends SentryAttachment {
-  ScreenshotAttachment()
-      : super.fromLoader(
-          filename: 'screenshot.png',
-          contentType: 'image/png',
-          loader: _createScreenshot,
-        );
+class ScreenshotAttachment implements SentryAttachment {
+  ScreenshotAttachment();
 
-  static FutureOr<Uint8List> _createScreenshot() async {
-    final _completer = Completer<Uint8List?>();
-    final instance = WidgetsBinding.instance;
+  @override
+  String attachmentType = SentryAttachment.typeAttachmentDefault;
+
+  @override
+  String? contentType = 'image/png';
+
+  @override
+  String filename = 'screenshot.png';
+
+  @override
+  FutureOr<Uint8List> get bytes async {
+    //return await createScreenshot() ?? Uint8List.fromList([]);
+    final instance = SchedulerBinding.instance;
     if (instance == null) {
-      _completer.complete(null);
-      return await _completer.future ?? Uint8List.fromList([]);
+      return Uint8List.fromList([]);
```

Comment:
maybe we allow returning null then? it does not make sense to return an array of nothing, unless we filter out before adding the attachment if the array is empty or not

Reply:
That would require all Attachments to allow returning null. Is that acceptable?